server/datastore/mysql/app_configs.go

```diff
@@ -36,6 +36,9 @@ func (ds *Datastore) GetCurrentTime(ctx context.Context) (time.Time, error) {
 }
 
 func (ds *Datastore) AppConfig(ctx context.Context) (*fleet.AppConfig, error) {
+	if ctxdb.IsPrimaryRequired(ctx) {
+		return appConfigDB(ctx, ds.writer(ctx))
+	}
```

Comment:
You don't need this check, the `ds.reader(...)` function was introduced for just that scenario, it will do the check itself and will return the `writer` if the primary is required: https://github.com/fleetdm/fleet/blob/5c398aafc08de395e7a8747a937923e5fdbb5b42/server/datastore/mysql/mysql.go#L101-L106